Source/VCGeneration/CheckerPool.cs

```diff
@@ -23,10 +23,6 @@ public CheckerPool(VCGenOptions options)
 
     public async Task<Checker> FindCheckerFor(Program program, Split? split, CancellationToken cancellationToken)
     {
-      if (disposed) {
-        throw new Exception("CheckerPool was already disposed");
-      }
-
       await checkersSemaphore.WaitAsync(cancellationToken);
```

Comment:
It seems that this task will still not be cancelled if the checkerpool is disposed of. Can you indicate me how it could be otherwise?

Reply:
The call `await checkersSemaphore.WaitAsync(cancellationToken);` throws an `ObjectDisposedException` if the `CheckerPool` disposed.